python/packages/nisar/workflows/stage_watermask.py

```diff
@@ -366,19 +382,99 @@ def download_watermask(polys, epsgs, outfile, version):
             xmin, ymin, xmax, ymax = poly.bounds
             translate_watermask(vrt_filename, outpath, xmin, xmax, ymin, ymax)
 
-        # Build vrt with downloaded watermasks
-        gdal.BuildVRT(outfile, watermask_list)
+        # Get the WATER description from the README.txt file using GDAL
+        in_readme_path = vrt_filename.replace(f'EPSG{epsg}.vrt', f'EPSG{epsg}/LICENSE.txt')
+        try:
+            water_descr = parse_readme_fields(in_readme_path)
+
+            # Build vrt with downloaded watermasks
+            vrt_dataset = gdal.BuildVRT(outfile, watermask_list)
+            # vrt_dataset.SetMetadataItem("dem_description", f'{dem_descr}')
+            for key, val in water_descr.items():
+                # # convert the bullet header to a tidy metadata key
+                # clean_key = key.lower().replace(" ", "_")
+                # vrt_dataset.SetMetadataItem(clean_key, val)
+                vrt_dataset.SetMetadataItem(key, val)
+        except Exception as e:
+            warnings.warn(
+                f"Could not read or parse README.txt at {in_readme_path} "
+                f"({e}). Proceeding without metadata."
+            )
+
     except Exception:
         errmsg = f'Failed to download NISAR WATERMASK {version} from s3 bucket. ' \
                  f'Maybe {version} is not currently supported.'
         raise ValueError(errmsg)
```

Comment:
Let's [chain](https://docs.python.org/3/tutorial/errors.html#exception-chaining) these exceptions so that the original exception message isn't lost, since the new error message doesn't give any useful information about what actually went wrong.

```suggestion
    except Exception as e:
        errmsg = f'Failed to download NISAR WATERMASK {version} from s3 bucket. ' \
                 f'Maybe {version} is not currently supported.'
        raise ValueError(errmsg) from e
```